scratch2/blocks.js

```diff
@@ -197,21 +200,14 @@ var BlockView = function(block) {
   this.children = block.children.map(newView)
   this.comment = this.comment ? newView(this.comment) : null
 
-  switch (this.info.category) {
-    case "music":
-      this.info.category = "sound"
-      break
-    case "video":
-      this.info.category = "sensing"
-      break
-    case "tts":
-    case "translate":
-    case "wedo":
-    case "ev3":
-    case "microbit":
-    case "makeymakey":
-      this.info.category = "extension"
-      break
+  if (ALIAS_EXTENSIONS.hasOwnProperty(this.info.category)) {
+    // handle aliases first
+    this.info.category = ALIAS_EXTENSIONS[this.info.category]
+  }
+  if (MOVED_EXTENSIONS.hasOwnProperty(this.info.category)) {
+    this.info.category = MOVED_EXTENSIONS[this.info.category]
+  } else if (EXTENSIONS.indexOf(this.info.category) > -1) {
```

Comment:
nit: It would be slightly better to (ab)use an object as a set here, so we can index it in constant time; that's almost certainly a premature optimisation though! :stuck_out_tongue:

Reply:
LOL, gonna do it